File: CampusAccessibilityBackend/CampusAccessibilityBackend/Core/Filters/ComplaintFilters.cs
namespace CampusAccessibilityBackend.Core.Filters
{
    public class ComplaintFilters
    {
        public string? Department { get; set; }
        public string? Category { get; set; }
        public string? Status { get; set; }
        public string? Keyword { get; set; }
        public string? SortOrder { get; set; }  // "newest" ή "oldest"
    }
}

File: campus-accessibility-frontend/src/types/complaintFilters.ts
//Φίλτρα
export interface ComplaintFilters {
    keyword?: string
    department?: string
    category?: string
    status?: string
    //για sorting στα φίλτρα
    sortOrder?: "newest" | "oldest"
}

//Pagination
export interface PaginatedResult<T> {
    data: T[]
    totalRecords: number
    pageNumber: number
    pageSize: number
    totalPages: number
}